staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -193,6 +193,16 @@ func ValidateMetrics(metrics Metrics, metricName string, expectedLabels ...strin
 	return nil
 }
 
+// GetMetricFamily returns MetricFamily whose name is equal to metricFamilyName from the metricFamily []*dto.MetricFamily
+func GetMetricFamily(metricFamily []*dto.MetricFamily, metricFamilyName string) *dto.MetricFamily {
```

Comment:
make it public due to we need to use it in the other places.

Reply:
Where? This shouldn't be public because it returns a `dto.MetricFamily` object which is a prometheus struct. If we expose it, it should return a wrapped variant. 

Also can you reference the usages? It's hard to say if this change makes sense without seeing the downstream changes.

Reply:
for example , we can use this function. see https://github.com/kubernetes/kubernetes/pull/89287/files#diff-3494b8fa37435f2a45fb7c430dfdb9c9R121.
```
pkg/volume/util/operationexecutor/operation_generator_test.go:	metricFamily := getMetricFamily(metricFamilyName)
pkg/volume/util/operationexecutor/operation_generator_test.go:func getMetricFamily(metricFamilyName string) *io_prometheus_client.MetricFamily {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:					func(mf *dto.MetricFamily) (*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:						mf = deepCopyMetricFamily(mf)
cluster/images/etcd-version-monitor/etcd-version-monitor.go:					func(mf *dto.MetricFamily) (*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:						mf = deepCopyMetricFamily(mf)
cluster/images/etcd-version-monitor/etcd-version-monitor.go:					func(mf *dto.MetricFamily) (*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:						mf = deepCopyMetricFamily(mf)
cluster/images/etcd-version-monitor/etcd-version-monitor.go:type rewriteFunc func(mf *dto.MetricFamily) (*dto.MetricFamily, error)
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func (m *monitorGatherer) Gather() ([]*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:	result := make([]*dto.MetricFamily, 0, len(exported)+len(custom))
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func (m *monitorGatherer) rewriteExportedMetrics(metrics map[string]*dto.MetricFamily) ([]*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:	results := make([]*dto.MetricFamily, 0, len(metrics))
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func scrapeMetrics() (map[string]*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func renameMetric(mf *dto.MetricFamily, name string) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func renameLabels(mf *dto.MetricFamily, nameMapping map[string]string) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func filterMetricsByLabels(mf *dto.MetricFamily, labelValues map[string]string) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func groupCounterMetricsByLabels(mf *dto.MetricFamily, names map[string]bool) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func deleteLabels(mf *dto.MetricFamily, names map[string]bool) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func identity(mf *dto.MetricFamily) (*dto.MetricFamily, error) {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:func deepCopyMetricFamily(mf *dto.MetricFamily) *dto.MetricFamily {
cluster/images/etcd-version-monitor/etcd-version-monitor.go:	r := &dto.MetricFamily{}
```
mosts of them return ([]*dto.MetricFamily, error).

for If we expose it, it should return a wrapped variant.  Agree.